### PR TITLE
store/tikv: set undetermined error if not recover from an rpc error.

### PR DESCRIPTION
**Pull request by disksing**
If encounter rpc error, record it in `sender`, then mark the error undetermined later.

### CODE REVIEW

File: store/tikv/2pc.go [hunk 1]
```diff
@@ -438,15 +438,18 @@ func (c *twoPhaseCommitter) commitSingleBatch(bo *Backoffer, batch batchKeys) er
 	// Under this circumstance,  we can not declare the commit is complete (may lead to data lost), nor can we throw
 	// an error (may lead to the duplicated key error when upper level restarts the transaction). Currently the best
 	// solution is to populate this error and let upper layer drop the connection to the corresponding mysql client.
-	isPrimary := bytes.Equal(batch.keys[0], c.primary())
+	sender := NewRegionRequestSender(c.store.regionCache, c.store.client)
+	errWrapper := func(err error) error {
+		if err != nil && bytes.Equal(batch.keys[0], c.primary()) && sender.rpcError != nil {
```
Comment (winkyao) [inline store/tikv/2pc.go]: why not keep `isPrimary := bytes.Equal(batch.keys[0], c.primary())` outside, and use isPrimary inside the closure.

Comment (disksing) [reply store/tikv/2pc.go]: If no error occurs, we don't have to check if it contains primary.

Comment (winkyao) [reply store/tikv/2pc.go]: got

File: store/tikv/2pc.go [hunk 2]
```diff
@@ -455,11 +458,11 @@ func (c *twoPhaseCommitter) commitSingleBatch(bo *Backoffer, batch batchKeys) er
 	if regionErr != nil {
 		err = bo.Backoff(BoRegionMiss, errors.New(regionErr.String()))
 		if err != nil {
-			return errors.Trace(err)
+			return errors.Trace(errWrapper(err))
 		}
 		// re-split keys and commit again.
 		err = c.commitKeys(bo, batch.keys)
-		return errors.Trace(err)
+		return errors.Trace(errWrapper(err))
```
Comment (coocood) [inline store/tikv/2pc.go]: `commitKeys` will call `commitSingleBatch` which should have wrapped the error already.

**Comment by coocood** [comment]
LGTM

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**